### ISSUE TITLE
Can't restart SSM Agent 2.2.916.0

### ISSUE THREAD

**Issue by ArturChe** [aws/amazon-ssm-agent#111]
Hi there,

I have updated EC2Config to a new version 4.9.2831 which installed new SSM Agent version 2.2.916.0 on Windows instance.

And it can not be started / restarted. On task manager I observing SSM Agent service status is Starting for half of hour. Then I tried to kill the process and start service manually. And it gave me the same behavior. There is nothing in logs for that period of time.

I don't know which additional information I can provide to clarify the issue. But previous version of EC2Config/SSM Agent were works fine.

**Comment by nehalaws** [comment]
Thank you for opening the issue, we are aware of this bug and we are working on releasing a new agent version that has a fix for this issue. Meanwhile, can you please check if the instance has the correct IAM role AmazonEC2RoleforSSM ? Having the proper credentials should fix the issue.

**Comment by ArturChe** [comment]
There were no AmazonEC2RoleforSSM role, and when I added it I have start SSM Agent successfully.
So should I modify my Terraform scripts to include that policy in IAM role for such kind of instances, or  it something that will be rollback to previous behavior?

**Comment by Kitsune-Fox** [comment]
@ArturChe, it has always been a requirement that instances managed with Systems Manager require a role with the correct policies attached.

**Comment by nehalaws** [comment]
Yes SSM agent features will work if the IAM role is present with the correct policy. We also published a fix for the issue where agent stuck in starting state. The fix is in version 2.3.13.0. Thank you!
